### PR TITLE
[typescript-sdk] Add documentation for typescript sdk

### PR DESCRIPTION
**Pull request by vchs1v**
## Motivation

This PR adds a documentation for Aptos typescript SDK in order to clarify its functionality

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

There is no need for test

### CODE REVIEW

**Comment by davidiw** [comment]
Nice PR, will leave it up to the main devs :)

**Comment by jjleng** [comment]
Really appreciate your PR. Here are some thoughts:
1) A lot of the class/API descriptions do not add much value. For example, it will great if you can provide more details, such as what a `accountAddress` looks like.
2) It is better to write good code comments and generate the class/API docs with something like TSDoc. 

BTW, we are working on adding more code comments that are friendly to TSDoc. You could expect to see more SDK docs soon.

**Comment by vchs1v** [comment]
@jjleng I have documented all methods with TSDoc. If you see some ways to improve it, lets discuss

File: ecosystem/typescript/sdk/src/token_client.ts [hunk 1]
```diff
@@ -165,8 +261,20 @@ export class TokenClient {
...
   }
 
-  // Retrieve the token's creation_num, which is useful for non-creator operations
-  async getTokenBalance(creator: MaybeHexString, collectionName: string, tokenName: string): Promise<number> {
+  /**
+   * Queries specific token from account's TokenStore
+   * @param creator Hex-encoded 16 bytes Aptos account address which created a token
+   * @param collectionName Name of collection, which holds a token
+   * @param tokenName Token name
+   * @returns Token in next form
```
Comment (CapCap) [inline ecosystem/typescript/sdk/src/token_client.ts:269]: is this correct? `in next form`

Comment (vchs1v) [reply ecosystem/typescript/sdk/src/token_client.ts:269]: I tried to pay attention to a Token object interface, that is below.

Comment (jjleng) [reply ecosystem/typescript/sdk/src/token_client.ts:269]: `@returns Token object in below format` ?

File: ecosystem/typescript/sdk/src/token_client.ts [hunk 2]
```diff
@@ -142,8 +216,30 @@ export class TokenClient {
...
   }
 
-  // Retrieve the token's creation_num, which is useful for non-creator operations
-  async getTokenData(creator: MaybeHexString, collectionName: string, tokenName: string): Promise<number> {
+  /**
+   * Queries token data from collection
+   * @param creator Hex-encoded 16 bytes Aptos account address which created a token
+   * @param collectionName Name of collection, which holds a token
+   * @param tokenName Token name
+   * @returns Token data in next form
```
Comment (jjleng) [inline ecosystem/typescript/sdk/src/token_client.ts:224]: `in below format`

File: ecosystem/typescript/sdk/src/token_client.ts [hunk 3]
```diff
@@ -128,6 +182,26 @@ export class TokenClient {
     return transactionHash;
   }
 
+  /**
+   * Queries collection data
+   * @param creator Hex-encoded 16 bytes Aptos account address which created a collection
+   * @param collectionName Collection name
+   * @returns Collection data in next form
```
Comment (jjleng) [inline ecosystem/typescript/sdk/src/token_client.ts:189]: `in below format`

File: ecosystem/typescript/sdk/src/aptos_client.ts [hunk 4]
```diff
@@ -225,14 +337,23 @@ export class AptosClient {
...
     const response = await this.transactions.submitTransaction(signedTxnRequest);
     raiseForStatus(202, response, signedTxnRequest);
     return response.data;
   }
 
-  /** Submits a signed transaction to the the endpoint that takes BCS payload. */
+  /**
+   * Submits a signed transaction to the the endpoint that takes BCS payload
+   * @param signedTxn A BCS transaction representation. It is a hex-encoded string prefixed with `0x`
+   * and fulfilled with two hex digits per byte
```
Comment (jjleng) [inline ecosystem/typescript/sdk/src/aptos_client.ts:354]: This is not accurate. Lets remove **It is a hex-encoded string prefixed with `0x` and fulfilled with two hex digits per byte**. The bytes are just BCS bytes, no hex-encoding. We need to update the openapi spec as well. I will do that.

**Review by jjleng — APPROVED** [review]
looks good. approved 🚀

The TSDoc for `submitSignedBCSTransaction` is inaccurate. Please fix before merging.

File: ecosystem/typescript/sdk/src/aptos_account.ts [hunk 5]
```diff
@@ -48,21 +73,47 @@ export class AptosAccount {
...
 
+  /**
+   * Derives account address, public key and private key
+   * @returns AptosAccountObject instance.
+   * @example An example of the returned AptosAccountObject object
+   * ```
+   * {
+   *    address: "0xe8012714cd17606cee7188a2a365eef3fe760be598750678c8c5954eb548a591",
+   *    publicKeyHex: "0xf56d8524faf79fbc0f48c13aeed3b0ce5dd376b4db93b8130a107c0a5e04ba04 ",
+   *    privateKeyHex: "0x009c9f7c992a06cfafe916f125d8adb7a395fca243e264a8e56a4b3e6accf940d2b11e9ece3049ce60e3c7b4a1c58aebfa9298e29a30a58a67f1998646135204"
```
Comment (jjleng) [inline ecosystem/typescript/sdk/src/aptos_account.ts:113]: // ts-ignore?

Comment (vchs1v) [reply ecosystem/typescript/sdk/src/aptos_account.ts]: Seems like eslint does not recognize `eslint-disable` within TSDoc comments. I decided to just use template string literal

**Pull request closed without merging**